email/resource_email.go

```diff
@@ -88,13 +100,33 @@ func resourceEmailCreate(d *schema.ResourceData, m interface{}) error {
 		from, []string{to}, []byte(msg))
 
 	if err != nil {
-		log.Printf("smtp error: %s", err)
-		return err
+		errorCode := extractStatusCode(err.Error())
+		// make tf configurable eventually? maybe?
+		maxRetries := 5
+		if errorCode == "421" {
+			for retries := maxRetries; retries > 0; retries-- {
+				// exit if error has been cleared
+				if err == nil {
+					break
+				}
+				// delay between retries
+				time.Sleep(10)
```

Comment:
I'm not sure if this is necessary, but here we could have some randomness so that retries don't all happen at the same time: https://en.wikipedia.org/wiki/Exponential_backoff#Collision_avoidance

Reply:
Just a thought, might be overengineering lol.

Reply:
i think this might another TF configurable option down the line depending on if we ever orchestrate multiple SMTP servers? 

Also if I am understanding correctly, this code runs on a resource basis, like as the resources are being created, and doesn't return until the resource was able to send the email or well, errors out, so in a way multiple resources will not be trying to use the server at the same time, so almost negligible chances for collision? is my understanding correct?

are we worried about using the gmail server concurrently?

Reply:
Terraform will try to parallelize the resource creation as much as possible. So when we have 20 new resources, Terraform will likely send all 20 requests at the same time. This is from observation, but I'm sure there's documentation on this somewhere.

The collision avoidance is to prevent too many concurrent requests to the same server.

Reply:
ahh makes sense, then it makes sense to add it. will incorporate